cmd/snap-confine/snap-confine.c

```diff
@@ -84,6 +85,18 @@ int main(int argc, char **argv)
 		die("need to run as root or suid");
 	}
 #endif
+
+	char *snap_context __attribute__ ((cleanup(sc_cleanup_string))) = NULL;
+	// Do no get snap context value if running a hook (we don't want to overwrite hook's SNAP_COOKIE)
+	if (!sc_is_hook_security_tag(security_tag)) {
+		struct sc_error *err
+		    __attribute__ ((cleanup(sc_cleanup_error))) = NULL;
+		snap_context = sc_cookie_get_from_snapd(snap_name, &err);
+		if (err != NULL) {
+			error("cannot get cookie: %s", sc_error_msg(err));
```

Comment:
I think it is fine to just print the message, errors already talk about the cookie.